worldengine/hdf5_serialization.py

```diff
@@ -0,0 +1,220 @@
+import h5py
+from worldengine.version import __version__
+from worldengine.biome import biome_name_to_index, biome_index_to_name
+from worldengine.world import World, Step
+import numpy
+
+
+def save_world_to_hdf5(world, filename):
+    f = h5py.File(filename, libver='latest', mode='w')
+
+    general_grp = f.create_group("general")
+    general_grp["worldengine_version"] = __version__
+    general_grp["name"] = world.name
+    general_grp["width"] = world.width
+    general_grp["height"] = world.height
+
+    elevation_grp = f.create_group("elevation")
+    elevation_ths_grp = elevation_grp.create_group("thresholds")
+    elevation_ths_grp["sea"] = world.elevation['thresholds'][0][1]
+    elevation_ths_grp["plain"] = world.elevation['thresholds'][1][1]
+    elevation_ths_grp["hill"] = world.elevation['thresholds'][2][1]
+    elevation_data = elevation_grp.create_dataset("data", (world.height, world.width), dtype=numpy.float)
+    elevation_data.write_direct(world.elevation['data'])
+
+    plates_data = f.create_dataset("plates", (world.height, world.width), dtype=numpy.uint16)
+    for y in range(world.height):
+        for x in range(world.width):
+            plates_data[y, x] = world.plates[y][x]
+
+    ocean_data = f.create_dataset("ocean", (world.height, world.width), dtype=numpy.bool)
+    ocean_data.write_direct(world.ocean)
+
+    sea_depth_data = f.create_dataset("sea_depth", (world.height, world.width), dtype=numpy.float)
+    sea_depth_data.write_direct(world.sea_depth)
+
+    if hasattr(world, 'biome'):
+        biome_data = f.create_dataset("biome", (world.height, world.width), dtype=numpy.uint16)
+        for y in range(world.height):
+            for x in range(world.width):
+                biome_data[y, x] = biome_name_to_index(world.biome[y][x])
+
+    if hasattr(world, 'humidity'):
+        humidity_grp = f.create_group("humidity")
+        humidity_quantiles_grp = humidity_grp.create_group("quantiles")
+        for k in world.humidity['quantiles'].keys():
+            humidity_quantiles_grp[k] = world.humidity['quantiles'][k]
+        humidity_data = humidity_grp.create_dataset("data", (world.height, world.width), dtype=numpy.float)
+        humidity_data.write_direct(world.humidity['data'])
```

Comment:
I think we should create some intermediate concepts such as "Layer" and perhaps having subclasses like "LayerWithQuantiles" or "LayerWithThresholds". In this way we could streamline many parts of the code, including serialization.